File: src/main/java/com/asap/server/domain/TimeBlock.java
package com.asap.server.domain;

import com.asap.server.domain.enums.TimeSlot;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Enumerated;
import jakarta.persistence.ManyToOne;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class TimeBlock extends AuditingTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ColumnDefault(value = "0")
    private int weight;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "available_date_id")
    private AvailableDate availableDate;

    @Enumerated(value = EnumType.STRING)
    private TimeSlot timeSlot;

    @OneToMany(mappedBy = "timeBlock")
    private List<TimeBlockUser> timeBlockUsers;

    public void addTimeBlockUsers(TimeBlockUser timeBlockUser) {
        if (timeBlockUsers == null) {
            timeBlockUsers = new ArrayList<>();
        }
        timeBlockUsers.add(timeBlockUser);
    }

    public void addWeight(int weight) {
        this.weight += weight;
    }
}

File: domain-db/src/main/kotlin/com/asap/server/domain/timeblock/TimeBlock.kt
package com.asap.server.domain.timeblock

import com.asap.server.domain.common.AuditingTimeEntity
import com.asap.server.domain.common.TimeSlot
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import org.hibernate.annotations.ColumnDefault

@Entity
class TimeBlock(
    id: Long? = null,
    weight: Int = 0,
    availableDateId: Long,
    timeSlot: TimeSlot
) : AuditingTimeEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = id
        protected set

    @ColumnDefault(value = "0")
    var weight = weight
        protected set

    @Column(nullable = false, name = "available_date_id")
    var availableDateId = availableDateId
        protected set

    @Enumerated(value = EnumType.STRING)
    var timeSlot = timeSlot
        protected set

    fun addWeight(weight: Int) {
        this.weight += weight
    }
}